src/main/java/pl/karolpat/serviceImpl/UserServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+package pl.karolpat.serviceImpl;
+
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.stereotype.Service;
+
+import pl.karolpat.entity.DailyIncome;
+import pl.karolpat.entity.ParkingMeter;
+import pl.karolpat.entity.User;
+import pl.karolpat.entity.Vehicle;
+import pl.karolpat.repository.UserRepo;
+import pl.karolpat.service.DailyIncomeService;
+import pl.karolpat.service.ParkingMeterService;
+import pl.karolpat.service.UserService;
+import pl.karolpat.service.VehicleService;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+	private UserRepo userRepo;
+	private VehicleService vehicleService;
+	private ParkingMeterService parkingMeterService;
+	private DailyIncomeService dailyIncomeService;
+
+	static private String NOT_VALID_USERNAME = "Username not available. Please select another one.";
+	static private String VIP_STATUS_CHANGE = "Cannot change vip status on user who uses praking.";
+
+	public UserServiceImpl(UserRepo userRepo, VehicleService vehicleService, ParkingMeterService parkingMeterService,
+			DailyIncomeService dailyIncomeService) {
+		this.userRepo = userRepo;
+		this.vehicleService = vehicleService;
+		this.parkingMeterService = parkingMeterService;
+		this.dailyIncomeService = dailyIncomeService;
+	}
+
+	@Override
+	public List<User> getAllUsers() {
+		return userRepo.findAll();
+	}
+
+	@Override
+	public User getOneById(long id) {
+		return userRepo.findOne(id);
+	}
+
+	@Override
+	public Object save(String username) {
+
+		User user = userRepo.findOneByUsername(username);
+		if (user != null) {
+
+			return NOT_VALID_USERNAME;
+		} else {
+
+			user = new User();
+			user.setUsername(username);
+			return userRepo.save(user);
+		}
+
+	}
+
+	@Override
+	public Object setUserAsVip(long id) {
+
+		User user = userRepo.findOne(id);
+
+		if (user.isStarted()) {
+			return VIP_STATUS_CHANGE;
+
+		} else {
+
+			user.setVip(true);
+			userRepo.save(user);
+			return user;
+		}
+
+	}
+
+	@Override
+	public List<User> findAllWhereVip(boolean vip) {
+		return userRepo.findAllByVip(true);
+	}
+
+	@Override
+	public User startParking(String number, User user) {
```

Comment:
gdzie jest zarządzanie transakcjami?

Reply:
Dobra, przyznam szczerze, że nie do końca ogarniałem tę kwestię i jak istotne to jest. Poszperałem trochę w materiałach i poszerzyłem wiedzę w temacie transakcji. W razie jakiegoś niepowodzenia wszelkie zmiany nie powinny wejść do bazy tylko zostać wycofane do stanu sprzed wprowadzania zmian.